presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -101,7 +106,7 @@ public Plan plan(Analysis analysis)
         }
 
         // make sure we produce a valid plan after optimizations run. This is mainly to catch programming errors
-        PlanSanityChecker.validate(root);
+        PlanSanityChecker.validate(root, session, metadata, sqlParser, symbolAllocator.getTypes());
```

Comment:
hmm... I don't like having to pass all these extra arguments around just to satisfy some checker's dependencies. This is the kind of thing we should be doing with Guice, where each checker is injected with the things it needs. However, that will be a much bigger change, so let's leave it as it is for now.